packages/react-instantsearch/src/core/createInstantSearchServer.js

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { version } from '../../package.json';
+import algoliasearchHelper, {
+  SearchResults,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import ReactDom from 'react-dom/server';
+import { getIndex, hasMultipleIndex } from './indexUtils';
+import { isEmpty } from 'lodash';
+import cis from './createInstantSearch';
+
+const searchParameters = [];
```

Comment:
Do you mean everything is synchronous server side wise in all situations of React server side rendering? We just need to ensure searchParameters, indexName and client are write/read in a synchronous mode and that there's no way for them to be erased/replaced at a moment we do not expect like:
- app1 server side render write variable
- app2 server sider render write variables
- app1 server side render tries to read variables
- app1 gets app2 variables
- 🏡 on 🔥